src/io/io_windows.zig

```diff
@@ -0,0 +1,1268 @@
+    /// This clock is not affected by discontinuous jumps in the system time, for example if the
+    /// system administrator manually changes the clock.
+    pub fn monotonic(self: *Self) u64 {
+        const m = blk: {
+            // Uses QueryPerformanceCounter() on windows due to it being the highest precision timer
+            // available while also accounting for time spent suspended by default:
+            // https://docs.microsoft.com/en-us/windows/win32/api/realtimeapiset/nf-realtimeapiset-queryunbiasedinterrupttime#remarks
+
+            // QPF need not be globally cached either as it ends up being a load from read-only
+            // memory mapped to all processed by the kernel called KUSER_SHARED_DATA (See "QpcFrequency")
+            // https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/ntddk/ns-ntddk-kuser_shared_data
+            // https://www.geoffchappell.com/studies/windows/km/ntoskrnl/inc/api/ntexapi_x/kuser_shared_data/index.htm
+            const qpc = os.windows.QueryPerformanceCounter();
+            const qpf = os.windows.QueryPerformanceFrequency();
+
+            // 10Mhz (1 qpc tick every 100ns) is a common QPF on modern systems.
+            // We can optimize towards this by converting to ns via a single multiply.
+            // https://github.com/microsoft/STL/blob/785143a0c73f030238ef618890fd4d6ae2b3a3a0/stl/inc/chrono#L694-L701
+            const common_qpf = 10_000_000;
+            if (qpf == common_qpf) break :blk qpc * (std.time.ns_per_s / common_qpf);
+
+            // Convert qpc to nanos using fixed point to avoid expensive extra divs and overflow.
+            const scale = (std.time.ns_per_s << 32) / qpf;
+            break :blk @as(u64, @truncate((@as(u96, qpc) * scale) >> 32));
+        };
+
+        // "Oops!...I Did It Again"
+        if (m < self.monotonic_guard) @panic("a hardware/kernel bug regressed the monotonic clock");
+        self.monotonic_guard = m;
+        return m;
+    }
+
+    /// A timestamp to measure real (i.e. wall clock) time, meaningful across systems, and reboots.
+    /// This clock is affected by discontinuous jumps in the system time.
+    pub fn realtime(_: *Self) i64 {
+        const kernel32 = struct {
+            extern "kernel32" fn GetSystemTimePreciseAsFileTime(
+                lpFileTime: *os.windows.FILETIME,
+            ) callconv(os.windows.WINAPI) void;
+        };
+
+        var ft: os.windows.FILETIME = undefined;
+        kernel32.GetSystemTimePreciseAsFileTime(&ft);
+        const ft64 = (@as(u64, ft.dwHighDateTime) << 32) | ft.dwLowDateTime;
+
+        // FileTime is in units of 100 nanoseconds
+        // and uses the NTFS/Windows epoch of 1601-01-01 instead of Unix Epoch 1970-01-01.
+        const epoch_adjust = std.time.epoch.windows * (std.time.ns_per_s / 100);
+        return (@as(i64, @bitCast(ft64)) + epoch_adjust) * 100;
+    }
+
+    pub fn tick(_: *Self) void {}
+};
+pub const system = os.windows;
+
+pub const Waker = struct {
```

Comment:
If this lives unrelated to the IO instance, you could use a Mutex+Condition.

If it's mean to live with an IO, it could reserve a completion_key from it and reuse the IO's iocp handle for posts.

Reply:
thanks for the pointer, i'll do the latter. this is meant to allow one thread to wakeup another thread that is otherwise waiting on I/O. on macOS, we use a machport and on linux, we use an eventfd. 

example: https://github.com/oven-sh/bun/blob/main/src/io/io_darwin.zig#L507-L578